apps/examples/stack_protection/stack_protection_main.c

```diff
@@ -127,28 +176,58 @@ int stack_prot_main(int argc, char *argv[])
 			ch = getchar();
 			i = 0;
 			if (ch == 'N' || ch == 'n') {
-				stack_prot_inlimit();
-				printf(" Test stack access within range using array size - Passed !!\n");
+				if (type_kthread) {
```

Comment:
We can reduce duplicated codes.
```
if (ch == 'R' || ...) {
   printf(...
   printf(...
   printf(...
   ch = getchar();

   if (exit) {
     do exit
   }

   if (type_kthread) {  <-- regardless of character, give it 
       ret_chk = ioctl(tc_fd, .....);
       if (....) {
         ...
       }
    } else {
        if (ch == 'N' ....
         .....
    }
```
But, need to concern invalid character.

Reply:
I have done something similar to your code snippet above.
Please check.
For invalid character, I had added check so that only valid character is passed as ioctl argument.